packages/core/src/render3/di.ts

```diff
@@ -344,6 +347,120 @@ export function injectAttributeImpl(tNode: TNode, attrNameToInject: string): str
   return null;
 }
 
+/**
+ * Returns a `TNode` of the location where the current `LView` is declared at.
+ *
+ * @param lView an `LView` that we want to find parent `TNode` for.
+ */
+function getDeclarationTNode(lView: LView): TNode|null {
+  const tView = lView[TVIEW];
+
+  // Return the declaration parent for embedded views
+  if (tView.type === TViewType.Embedded) {
+    ngDevMode && assertDefined(tView.declTNode, 'Embedded TNodes should have declaration parents.');
+    return tView.declTNode;
+  }
+
+  // Components don't have `TView.declTNode` because each instance of component could be
+  // inserted in different location, hence `TView.declTNode` is meaningless.
+  // Falling back to `T_HOST` in case we cross component boundary.
+  if (tView.type === TViewType.Component) {
+    return lView[T_HOST];
+  }
+
+  // Remaining TNode type is `TViewType.Root` which doesn't have a parent TNode.
+  return null;
+}
+
+/**
+ * Calculates the parent TNode and matching LView for a given TNode.
+ *
+ * Determines the node host or walks the declaration tree until it finds
+ * the next valid parent node, if it exists, and sets that.
+ *
+ * @param tNode a `TNode` that we want to find the parent of.
+ * @param lView an `LView` that corresponds to the tNode param.
+ * @param flags an `InjectFlags` that we check to skip @Host.
+ */
+function calculateParentTNodeAndLView(tNode: TNode, lView: LView, flags: InjectFlags): void {
```

Comment:
Based on a function name it feels like it should return something (the result of the calculation). May be we can slightly update it to mention that we also *set* the value (thus no return value):
```suggestion
function calculateAndSetParentTNodeAndLView(tNode: TNode, lView: LView, flags: InjectFlags): void {
```

Reply:
This logic was moved inside enterDI in the fixup commit. So need to rename anymore.